superset/models/core.py

```diff
@@ -639,9 +639,19 @@ def sqlalchemy_uri_decrypted(self) -> str:
     def sql_url(self) -> str:
         return f"/superset/sql/{self.id}/"
 
-    def get_perm(self) -> str:
+    @hybrid_property
+    def perm(self) -> str:
         return f"[{self.database_name}].(id:{self.id})"
 
+    @perm.expression  # type: ignore
+    def perm(cls) -> str:  # pylint: disable=no-self-argument
+        return (
+            "[" + cls.database_name + "].(id:" + expression.cast(cls.id, String) + ")"
+        )
+
+    def get_perm(self) -> str:
+        return self.perm  # type: ignore
```

Comment:
Is there some reason we want to keep this around and not just reference `perm` wherever needed?

Reply:
@villbro it’s required [here](https://github.com/apache/incubator-superset/blob/550e78ff7c02491717960d39ef0bb861aba0f977/superset/security/manager.py#L801). Note the `set_perm` callback is still required as it has other logic besides merely setting the `perm` column. For the `dbs` and `clusters` table `perm` and `get_perm` are equivalent and hence no database record will be updated (this has always been the case for the `clusters` table).

Reply:
it is also used here: https://github.com/apache/incubator-superset/blob/550e78ff7c02491717960d39ef0bb861aba0f977/superset/security/manager.py#L820
https://github.com/apache/incubator-superset/blob/8e23d4f369f35724b34b14def8a5a8bafb1d2ecb/superset/models/core.py#L659
logic that keeps FAB security in sync with the other models